src/controllers/generalController.ts

```diff
@@ -0,0 +1,32 @@
+import express from "express";
+import { DishModel } from "../models/dish";
+import { RestaurantModel } from "../models/restaurant";
+import { ChefModel } from "../models/chef";
+
+export const getResultsBySearch = async (req: express.Request, res: express.Response) => {
```

Comment:
if we're using post api call the terminology of get is not correct, try to change it to searchByInput or similar

Reply:
fixed